### PR TITLE
Improve Lido reverted-tx-watcher bot

### PR DESCRIPTION
**Pull request by larmork**
Low and medium severity alerts added for high gas usage in reverted tx and accordingly for large amount of reverts per day by one EOA

### CODE REVIEW

**Review by dgusakov — CHANGES_REQUESTED** [review]

File: reverted-tx-watcher/src/agent-reverted-tx.ts [hunk 1]
```diff
@@ -74,36 +86,70 @@ async function handleRevertedTx(
...
+      address, 
+      txEvent
+    )
+  )
+}
 }
 
-function etherscanLink(txHash: string): string {
+function startTimer() {
+  setInterval(resetDictionary, 24 * 60 * 60 * 1000);
```
[the remark this answers is not in the document]
Comment (larmork) [reply reverted-tx-watcher/src/agent-reverted-tx.ts]: Done!

File: reverted-tx-watcher/src/findings.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        type: FindingType.Suspicious,
+        metadata: {
+          sender: txEvent.from,
+          contract: address,
+          gas: gasUsed.toString(),
+        },
+      })
+    };
+
+  export const createLowSeverityFinding = (
```
[the remark this answers is not in the document]
Comment (larmork) [reply reverted-tx-watcher/src/findings.ts]: Fixed

File: reverted-tx-watcher/src/findings.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        type: FindingType.Suspicious,
+        metadata: {
+          sender: txEvent.from,
+          contract: address,
+          gas: gasUsed.toString(),
+        },
+      })
+    };
+
+  export const createMediumSeverityFinding = (
```
[the remark this answers is not in the document]
Comment (larmork) [reply reverted-tx-watcher/src/findings.ts]: Fixed

File: reverted-tx-watcher/src/findings.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        alertId: "REVERTED-TX",
+        severity: FindingSeverity.Medium,
+        type: FindingType.Suspicious,
+        metadata: {
+          sender: txEvent.from,
+          contract: address,
+          gas: gasUsed.toString(),
+        },
+      })
+    };
```
[the remark this answers is not in the document]
Comment (larmork) [reply reverted-tx-watcher/src/findings.ts]: Already added

File: reverted-tx-watcher/src/agent-reverted-tx.ts [hunk 5]
```diff
@@ -13,11 +11,25 @@ import {
...
 
+export const name = "RevertedTxWatcher";
+export const HIGH_GAS_THRESHOLD = "600000";
+export const MAX_REVERTION_PER_DAY = 15
+let revertedTxPerEOA: RevertDictionary = {};
 let contracts: LocatorContracts;
 let addresses: [string, string][] = [];
+export let gasUsed = ethers.BigNumber.from(0);
+
+startTimer()
```
[the remark this answers is not in the document]
Comment (larmork) [reply reverted-tx-watcher/src/agent-reverted-tx.ts]: Done

**Review by dgusakov — APPROVED** [review]

**Review by sergeyWh1te — APPROVED** [review]
It'd be great to add eslint + prettier rules for the project.

**Review by dgusakov — APPROVED** [review]

**Review by sergeyWh1te — APPROVED** [review]

**Pull request merged**